django/core/handlers/base.py

```diff
@@ -120,10 +155,11 @@ def get_response(self, request):
         try:
             response = None
             # Apply request middleware
-            for middleware_method in self._request_middleware:
-                response = middleware_method(request)
-                if response:
-                    break
+            if settings.MIDDLEWARES is None:
```

Comment:
Rather than having these two conditionals based on settings, what if we extract a new method that only does view middleware, calls the view, and template response middleware? Then that method can be the baseline "handler" in the new system, and we can have a wrapper method for the old system that does request middleware, calls the new method, then does response and exception middleware.

For the sake of discussion, let's say we call the new method `_get_response` and the old one `_get_response_with_old_middleware`.

When the old system is in use, `load_middleware could just do `self._middleware_chain = self._get_response_with_old_middleware`; that way the top-level `get_response` wouldn't need any conditionals at all, and all the old-vs-new logic would be encapsulated in `load_middleware`.

I think this would make for a clearer separation of old vs new, and that would make the end of the deprecation path simpler too, since the only things that would change would be a) cut out half of `load_middleware`, and b) remove `_get_response_with_old_middleware`.